File: library/src/main/java/com/shxhzhxx/sdk/utils/ToastUtils.java
package com.shxhzhxx.sdk.utils;

import android.content.Context;
import android.widget.Toast;

import java.lang.ref.WeakReference;

public abstract class ToastUtils {
    private static Toast mToast;
    private static WeakReference<Context> mContextRef;

    public static void init(Context context) {
        mContextRef = new WeakReference<>(context);
    }

    public static void show(String msg) {
        Context context = mContextRef.get();
        if (context == null)
            return;
        if (mToast != null) {
            mToast.cancel();
        }
        mToast = Toast.makeText(context, msg, Toast.LENGTH_SHORT);
        mToast.show();
    }
}

File: library/src/main/java/com/shxhzhxx/sdk/utils/ToastUtils.kt
package com.shxhzhxx.sdk.utils

import android.content.Context
import android.widget.Toast

import java.lang.ref.WeakReference

private var toast: Toast? = null
private lateinit var contextRef: WeakReference<Context>

fun initToast(context: Context) {
    contextRef = WeakReference(context)
}

fun toast(msg: String) {
    val context = contextRef.get() ?: return
    toast?.cancel()
    toast = Toast.makeText(context, msg, Toast.LENGTH_SHORT).also { it.show() }
}
